rootless-install.sh

```diff
@@ -0,0 +1,264 @@
+#!/bin/sh
+
+# This script is meant for quick & easy install via:
+#   $ curl -fsSL https://rootless.docker.com -o get-docker.sh
+#   $ sh get-docker.sh
+#
+# NOTE: Make sure to verify the contents of the script
+#       you downloaded matches the contents of install.sh
+#       located at https://github.com/docker/docker-install
+#       before executing.
+#
+# Git commit from https://github.com/docker/docker-install when
+# the script was uploaded (Should only be modified by upload job):
+SCRIPT_COMMIT_SHA=UNKNOWN
+
+# This script should be run with an unprivileged user and install/setup Docker under $HOME/bin/.
+
+set -e
+
+init_vars() {
+	BIN="$HOME/bin"
+	DAEMON=dockerd
+	SYSTEMD=
+	if systemctl --user daemon-reload >/dev/null 2>&1; then
+		SYSTEMD=1
+	fi
+}
+
+checks() {
+	# OS verification: Linux only, point osx/win to helpful locations
+	case "$(uname)" in
+	Linux)
+		;;
+	*)
+		>&2 echo "Rootless Docker cannot be installed on $(uname)"; exit 1
+		;;
+	esac
+
+	# User verification: deny running as root (unless forced?)
+	if [ "$(id -u)" = "0" ]; then
+		>&2 echo "Refusing to install rootless Docker as the root user"; exit 1
+	fi
+
+	# HOME verification
+	if [ ! -d "$HOME" ]; then
+		>&2 echo "Aborting because HOME directory $HOME does not exist"; exit 1
+	fi
+
+	if [ -d "$BIN" ]; then
+		if [ ! -w "$BIN" ]; then
+			>&2 echo "Aborting because $BIN is not writable"; exit 1
+		fi
+	else
+		if [ ! -w "$HOME" ]; then
+			>&2 echo "Aborting because HOME (\"$HOME\") is not writable"; exit 1
+		fi
+	fi
+
+	# Existing rootful docker verification
+	if [ -w /var/run/docker.sock ]; then
+		>&2 echo "Aborting because rootful Docker is running and accessible"; exit 1
```

Comment:
Why?

Reply:
Even rootful Docker is accessible, rootless Docker is still useful for hardening purpose.

Reply:
To avoid mistakes on beginner users. We could add a `FORCE` override for this. Advanced users can use the manual steps as well (actually we should print out the link to manual steps in this script).

Reply:
Can we show `y/N` prompt rather than aborting? Can be a follow-up PR.

Reply:
+1 on leaving prompts to follow-up